train.py

```diff
@@ -212,14 +231,48 @@ def main():
     if rank == 0:
         logging.info('Beginning training')
     fit_verbose = 1 if (args.verbose and rank==0) else 2
-    model.fit(datasets['train_dataset'],
-              steps_per_epoch=datasets['n_train_steps'],
-              epochs=data_config['n_epochs'],
-              validation_data=datasets['valid_dataset'],
-              validation_steps=datasets['n_valid_steps'],
-              callbacks=callbacks,
-              initial_epoch=initial_epoch,
-              verbose=fit_verbose)
+
+    #cray added
+    #timemory.enable_signal_detection()
+    #timemory.settings.width = 12
+    #timemory.settings.precision = 6
+
+    #with profile(["wall_clock", "user_clock", "system_clock", "cpu_util",
+    #              "peak_rss", "thread_cpu_clock", "thread_cpu_util"]):
+    #id = timemory.start_mpip()
+
+    #with marker(['wall_clock','cpu_util','peak_rss','cpu_roofline_flops','gpu_roofline_flops','user_mpip_bundle','read_bytes', 'written_bytes'], key="marker_ctx_manager"):
+
+    #with marker(['wall_clock','cpu_util','peak_rss','cpu_roofline_flops','user_mpip_bundle','read_bytes', 'written_bytes'], key="marker_ctx_manager"):
+
+    #with profile(['wall_clock','cpu_util','peak_rss','cpu_roofline_sp_flops','gpu_roofline_flops','user_mpip_bundle',
+    #                'read_bytes', 'written_bytes'], flat=True, timeline=False):
+
+    #components = ['wall_clock', 'cpu_util', 'peak_rss','read_bytes', 'written_bytes','thread_cpu_util','user_mpip_bundle','thread_cpu_clock']
+    timemory.settings.flat_profile = False
+    timemory.settings.timeline_profile = False
+
+    components = ['wall_clock','peak_rss','read_bytes','written_bytes']
```

Comment:
You should add in the `'user_global_bundle'` here. It will essentially be zero overhead and it will allow you to add other components on-the-fly without and modifications to the script, e.g. `export TIMEMORY_GLOBAL_COMPONENTS="papi_vector, virtual_memory, gperftools_cpu_profiler"`